### PR TITLE
JI769932-P2022-1797-fix-SearchBar

### PR DESCRIPTION
**Pull request by Reden4005**
FIX - User can submit an empty field or less than 3 characters in the search bar

### CODE REVIEW

File: libs/shared/src/components/search-bar/search-bar.tsx [hunk 1]
```diff
@@ -16,6 +16,12 @@ export const SearchBar: React.FC = () => {
     setIsOpen(prevState => !prevState);
   };
 
+  const handleEnter = (event: KeyboardEvent) => {
```
Comment (pawelwojdas) [inline libs/shared/src/components/search-bar/search-bar.tsx:19]: I would change name to handleEnterKey

File: libs/shared/src/components/search-bar/search-drawer/index.tsx [hunk 2]
```diff
@@ -21,9 +21,10 @@ interface SearchDrawerProps {
   searchKey: string;
   toResult: AppRouteType;
   toItem: AppRouteType;
+  onKeyPress: (event: KeyboardEvent) => void;
 }
 
-export const SearchDrawer: React.FC<SearchDrawerProps> = ({ open, onClose, searchKey, toResult, toItem }) => {
+export const SearchDrawer: React.FC<SearchDrawerProps> = ({ open, onClose, searchKey, toResult, toItem, onKeyPress }) => {
   const { t } = useTranslation(TranslationNamespace.Feedback);
   const { i18n } = useTranslation();
```
Comment (krysia1) [inline libs/shared/src/components/search-bar/search-drawer/index.tsx:29]: Why not get those in one const?

File: libs/shared/src/components/search-bar/search-bar.tsx (hunk 1 above)
Comment (Reden4005) [reply libs/shared/src/components/search-bar/search-bar.tsx:19]: ok

File: libs/shared/src/components/search-bar/search-drawer/index.tsx (hunk 2 above)
Comment (Reden4005) [reply libs/shared/src/components/search-bar/search-drawer/index.tsx:29]: You're right

File: libs/shared/src/components/search-bar/search-bar.tsx [hunk 3]
```diff
@@ -16,6 +16,12 @@ export const SearchBar: React.FC = () => {
     setIsOpen(prevState => !prevState);
```
[the remark this answers is not in the document]
Comment (Reden4005) [reply libs/shared/src/components/search-bar/search-bar.tsx:16]: ok

**Comment by piotrrawski** [comment]
LGTM

**Comment by krysia1** [comment]
LGTM 🎉

**Review by pawelwojdas — COMMENTED** [review]
LGTM

**Comment by kacperChlon** [comment]
Please correct branch name

File: libs/shared/src/components/search-bar/search-bar.tsx [hunk 4]
```diff
@@ -10,12 +11,18 @@ import { useSearchConfig } from "./use-search-config";
...
   const { searchKey, allResultsPage, singleResultPage } = useSearchConfig();
   const [isOpen, setIsOpen] = useState(false);
-  const { t } = useTranslation();
+  const { t } = useTranslation(TranslationNamespace.Feedback);
 
   const handleClick = () => {
     setIsOpen(prevState => !prevState);
   };
 
+  const handleEnterKey = (event: KeyboardEvent) => {
```
Comment (kacperChlon) [inline libs/shared/src/components/search-bar/search-bar.tsx:20]: handleKeyPress

File: libs/shared/src/components/search-bar/search-bar.tsx [hunk 5]
```diff
@@ -10,12 +11,18 @@ import { useSearchConfig } from "./use-search-config";
...
+    if (event.key === "Enter") {
```
Comment (kacperChlon) [inline libs/shared/src/components/search-bar/search-bar.tsx:21]: please move "Enter" to some shared enum, for instance:

```
enum KeyCode {
 Enter = 'Enter'
}
```

**Review by kacperChlon — APPROVED** [review]

File: libs/shared/src/components/search-bar/search-bar.tsx (hunk 4 above)
Comment (Reden4005) [reply libs/shared/src/components/search-bar/search-bar.tsx:20]: Changed.

File: libs/shared/src/components/search-bar/search-bar.tsx (hunk 5 above)
Comment (Reden4005) [reply libs/shared/src/components/search-bar/search-bar.tsx:21]: Ok

File: libs/shared/src/components/search-bar/search-bar.tsx [hunk 6]
```diff
@@ -1,21 +1,32 @@
...
+import { KeyboardEvent, useState } from "react";
 import { useTranslation } from "react-i18next";
 
+import { TranslationNamespace } from "../../types";
 import { BaseButton, ButtonType } from "../base-button";
 import { SearchDrawer } from "./search-drawer";
 import * as Styled from "./styled";
 import { useSearchConfig } from "./use-search-config";
 
+enum KeyCode {
```
Comment (Adys23) [inline libs/shared/src/components/search-bar/search-bar.tsx:11]: maybe it is worth to move it to a separate file in `shared/src/types`, as it is very generic 😄 and might be used elsewhere.

Comment (Reden4005) [reply libs/shared/src/components/search-bar/search-bar.tsx:11]: ok

**Review by aniasgit — APPROVED** [review]

**Review by Adys23 — APPROVED** [review]

**Pull request merged**